feat golang: enabled and fixed data tags, added comments from logs, added -i flag

TertiumOrganum1 opened the pull request:
## Description
Anyway, I've managed to fix the Golang generator: I added comments from descriptions, enabled them in the Go helper, fixed the data tags by adding "required", added a flag to include comments in Go, and included the description preset in modelina-cli. I'm not sure why, but modelina-cli refused to accept the file parameter from the command line (even after packing it into a tarball), so I had to add an optional -i flag to specify the input path.


## Related Issue
[FEATURE] Golang generator should include comments from parameter descriptions #2116

## Checklist
- [ ] The code follows the project's coding standards and is properly linted (`npm run lint`): yes, but two prettier issues are in conflict with default vscode ts formatter, so I left as is.
- [ ] Tests have been added or updated to cover the changes: not yet
- [ ] Documentation has been updated to reflect the changes: yes
- [ ] All tests pass successfully locally.(`npm run test`): not yet 

## Additional Notes

I haven't been able to build the CLI correctly yet: it can't use the modelina library directly from the source (or via npm link). Therefore, I had to run npm run build:prod for modelina, copy its lib folder to the modelina-cli node_modules directory (after the initial npm install), and only then was I able to build modelina-cli and run it. Any advice on how to do this correctly would be appreciated. For now, I'll create a PR just for you to take a look.

jonaslagoni commented on src/generators/go/renderers/StructRenderer.ts:
```diff
@@ -20,27 +20,21 @@ export class StructRenderer extends GoRenderer<ConstrainedObjectModel> {
-      `${this.model.name} represents a ${this.model.name} model.`
-    );
-
+    const doc = '';
```
You can remove it completely if it's no longer in use 🙂

jonaslagoni reviewed (changes requested):
Remember to add the new feature to the Go documentation https://github.com/TertiumOrganum1/modelina/blob/next/docs/languages/Go.md you can just use this as guideline: https://github.com/TertiumOrganum1/modelina/blob/next/docs/languages/TypeScript.md#rendering-comments-from-description-and-example-fields

We also need a single test ensuring that the preset works as expected, just take the one from TypeScript and change it to use Go 🙂 https://github.com/TertiumOrganum1/modelina/blob/next/test/generators/typescript/preset/DescriptionPreset.spec.ts

TertiumOrganum1 commented:
> Remember to add the new feature to the Go documentation https://github.com/TertiumOrganum1/modelina/blob/next/docs/languages/Go.md you can just use this as guideline: https://github.com/TertiumOrganum1/modelina/blob/next/docs/languages/TypeScript.md#rendering-comments-from-description-and-example-fields
> 
> We also need a single test ensuring that the preset works as expected, just take the one from TypeScript and change it to use Go 🙂 https://github.com/TertiumOrganum1/modelina/blob/next/test/generators/typescript/preset/DescriptionPreset.spec.ts

Sorry, I don't want it wants from me. What it mean by duplication? Take a look and go examples - the have the same index.spec.ts! But my test is duplicated? 
Could you please explain what does it want or even fix it? Or should I just delete my tests?

jonaslagoni commented:
/rtm

jonaslagoni reviewed (changes requested):
Looks like you might be missing to export the preset :v:

TertiumOrganum1 commented:
> Looks like you might be missing to export the preset ✌️
What do you mean? 
in go/presets/index.ts: export * from './DescriptionPreset';
in go/presets/DescriptionPreset.ts: export const GO_DESCRIPTION_PRESET: GoPreset = . . . 
And on my PC modelina works as expected.
As far as I can see everything done similar to typescript generator.

TertiumOrganum1 commented:
/ptal @jonaslagoni  Hi, please advise what to do with my PR here. It didn't pass tests as modelina-cli uses modelina as npm package, but I've fixed both. Locally it all works but not in your CI. So should we split it into two PRs? If fix to modelina will be merged, how long should we wait to make PR to modelina-cli? Or maybe there is a way to disable some checks on this particular PR? How do you make changes in modelina&cli anyways? npm dependency makes them impossible

jonaslagoni commented:
> @jonaslagoni Hi, please advise what to do with my PR here. It didn't pass tests as modelina-cli uses modelina as npm package, but I've fixed both. Locally it all works but not in your CI. So should we split it into two PRs? If fix to modelina will be merged, how long should we wait to make PR to modelina-cli? Or maybe there is a way to disable some checks on this particular PR? How do you make changes in modelina&cli anyways? npm dependency makes them impossible

Yea, you right, completely forgot about that...

Split it up and as soon as it's released we can merge the CLI changes 🙂

jonaslagoni commented:
@all-contributors please add @TertiumOrganum1 for code, docs, test
